src/components/tinyangel/TinyAngel.tsx

```diff
@@ -0,0 +1,244 @@
+        relevantRewards = relevantRewards.map((el: any) => {
+            return {
+                ...el,
+                amount: parseInt(el.amount)
+            }
+        })
+
+        setRewardState({ totalRewards: relevantRewards })
+    }
+
+    const refetchUserState = () => {
+        tinyBalanceSetter()
+        rewardStateSetter()
+    }
+
+    const onDonate = async () => {
+        if ( nativeWalletState.tinyBalances.length === 0 ) {
+            alert("You don't have any tiny balances to donate")
+            return;
+        }
+
+        const balancesObj = nativeWalletState.tinyBalances.reduce((obj, el) => {
+            return Object.assign(obj, { [el.denom]: el.amount })
+        }, {})
+
+        const msgs = donateTinyAmount(user_address, ANGEL_PROTO_ADDRESS_BOMBAY, balancesObj)
+        setMsgs(msgs);
+    }
+
+    const onRewardDonate = () => {
+        const msgs: any[] = claimReward(user_address, rewardState.validatorAddresses)
+
+        if ( rewardState.totalRewards.length !== 0 ) {
+            const tinyRewardsObj = rewardState.totalRewards.reduce((obj, el) => {
+                return Object.assign(obj, { [el.denom]: el.amount })
+            }, {})
+
+            const [ sendRewardsToAngelMsg ] = donateTinyAmount(user_address, ANGEL_PROTO_ADDRESS_BOMBAY, tinyRewardsObj);
+            msgs.push(sendRewardsToAngelMsg);
+        }
+
+        setMsgs(msgs)
+    }
+
+    useEffectOnce(() => {
+        ;(async () => {
+            /* UST Swaprate to calculate all denominations into appropriate unified tiny amount limit */
+            const { data: swaprates } = await axios.get(ustSwapRateQuery);
+            const swapMap = swaprates.reduce((map: any, obj: any) => { map.set(obj.denom, obj.swaprate); return map; }, new Map);
+            setUstSwapRateMap(swapMap);
+
+            const [delegations] = await LCD.staking.delegations(user_address)
+            const validatorAddresses = delegations.map(e => e.validator_address);
+            setRewardState({ validatorAddresses })
+        })();
+    })
+
+    useEffect(() => {
+        if ( !user_address || ustSwapRateMap === undefined ) return;
+
+        const getTinyBalances = setTimeout(tinyBalanceSetter, 200)
+        return () => clearTimeout(getTinyBalances);
+    }, [ user_address, nativeWalletState.tinyValue, ustSwapRateMap ]);
+
+    useEffect(() => {
+        if ( !user_address ) return;
+
+        const getTotalRewards = setTimeout(rewardStateSetter, 200);
+        return () => clearTimeout(getTotalRewards);
+    }, [ user_address, rewardState.tinyReward ])
+
+    return (
+        <>
+        <Container sx={{maxWidth: '1200px', padding: '10px'}}>
+            <Grid container spacing={2}>
+                <Grid item xs={12} sm={6}>
+                    <Paper elevation={3} style={{height: 'fit-content'}}>
+                        <Stack spacing={1} sx={{padding: '30px'}}>
```

Comment:
can we add a small description of the feature and what it does somewhere in here?

Reply:
done